CMakeLists.txt

```diff
@@ -1477,24 +1484,34 @@ else()
     # #2536: cannot specify explicit initializer for arrays
     hpx_add_compile_flag_if_available(-wd2536)
   endif()
-
+  
+  
   # rdtsc is an x86 instruction that reads the value of a CPU time stamp
   # counter. rdtscp is an extension to rdtsc. The difference is that rdtscp is
   # a serializing instruction.
   hpx_cpuid("rdtsc" HPX_WITH_RDTSC
     DEFINITIONS HPX_HAVE_RDTSC)
 
+ # One can not assume if RDTSCP is available on the hardware
+ # of the build infrastructure, that it will be available on 
+ # all potential target hardware, see Issue  #3575
+ if(NOT $HPX_WITH_BUILD_DISTRIBUTION)
+
   # XeonPhi's do not support RDTSCP
   if(NOT ("${HPX_PLATFORM_UC}" STREQUAL "XEONPHI"))
     hpx_cpuid("rdtscp" HPX_WITH_RDTSCP
       DEFINITIONS HPX_HAVE_RDTSCP)
   endif()
 
+ endif()
+
   if(NOT HPX_WITH_RDTSC AND NOT HPX_WITH_RDTSCP)
     hpx_warn("Neither rdtsc nor rdtscp is available; some performance counters may report incorrect results")
   endif()
 endif()
 
+
```

Comment:
nitpicking: There are some unrelated whitespace changes, which is slightly bad style.

Reply:
Removed